legend-engine-authentication-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/authentication/AuthenticationLexerGrammar.g4

```diff
@@ -0,0 +1,36 @@
+lexer grammar AuthenticationLexerGrammar;
+
+import CoreLexerGrammar;
+
+// ---------------- OAuth Authentication
+
+OAUTH_AUTHENTICATION:                                       'OAuth';
+GRANT_TYPE:                                                 'grantType';
+CLIENT_ID:                                                  'clientId';
+CLIENT_SECRET_VAULT_REFERENCE:                              'clientSecretVaultReference';
```

Comment:
clientSecret to take a Vault Credential and not directly a vault reference 
clientSecret : CredentialVault { ref = "ref1"}